Sources/MuxPlayerSwift/FairPlay/ContentKeySessionDelegate.swift

```diff
@@ -5,125 +5,155 @@
              previous content key request.
              */
-        case AVContentKeyRequest.RetryReason.receivedResponseWithExpiredLease:
-            shouldRetry = true
-            
+        case .receivedResponseWithExpiredLease:
+            return true
+
             /*
              Indicates that the content key request should be retried because an obsolete key response was set on the previous
              content key request.
              */
-        case AVContentKeyRequest.RetryReason.receivedObsoleteContentKey:
-            shouldRetry = true
-            
+        case .receivedObsoleteContentKey:
+            return true
+
         default:
-            break
+            return false
         }
-        
-        return shouldRetry
     }
     
     // MARK: Logic
     
     func parsePlaybackId(fromSkdLocation uri: URL) -> String? {
         // pull the playbackID out of the uri to the key
-        let urlComponents = URLComponents(url: uri, resolvingAgainstBaseURL: false)
-        guard let urlComponents = urlComponents else {
+        guard let urlComponents = URLComponents(
+            url: uri,
+            resolvingAgainstBaseURL: false
+        ) else {
             // not likely
-            print("!! Error: Cannot Parse URI")
-            return nil
-        }
-        let playbackID = urlComponents.findQueryValue(key: "playbackId")
-        guard let playbackID = playbackID else {
-            print("!! Error: URI [\(uri)] did not have playbackId!")
+            logger.debug("\(#function) Error: cannot parse key URL [\(uri)]")
             return nil
         }
-        print("|| PlaybackID from \(uri) is \(playbackID)")
-        return playbackID
+
+        return urlComponents.findQueryValue(
+            key: "playbackId"
+        )
     }
     
     func handleContentKeyRequest(request: any KeyRequest) {
-        print("<><>handleContentKeyRequest: Called")
+        logger.debug(
+            "Called \(#function)"
+        )
+
+        guard let sessionManager = self.sessionManager else {
+            // TODO: Should this also invoke `processContentKeyResponseError`?
+            logger.debug("Missing session manager")
+            return
+        }
+
         // for hls, "the identifier must be an NSURL that matches a key URI in the Media Playlist." from the docs
-        guard let keyURLStr = request.identifier as? String,
-              let keyURL = URL(string: keyURLStr),
-              let assetIDData = keyURLStr.data(using: .utf8)
+        guard let requestIdentifierString = request.identifier as? String,
+              let mediaPlaylistKeyURL = URL(string: requestIdentifierString),
+              let utfEncodedRequestIdentifierString = requestIdentifierString.data(using: .utf8)
         else {
-            print("request identifier was not a key url, this is exceptional for hls")
+            // TODO: Should this also invoke `processContentKeyResponseError`?
```

Comment:
If it happened, it would be an error for sure. For HLS streams, it will always be a string that is a URL ([docs](https://developer.apple.com/documentation/avfoundation/avcontentkeyrequest/2799205-identifier), which I should have linked in a comment)

Reply:
Got it. Will add in separate follow-on PR.